io.openems.edge.common/src/io/openems/edge/common/type/CircularTreeMap.java

```diff
@@ -0,0 +1,32 @@
+package io.openems.edge.common.type;
+
+import java.util.Iterator;
+import java.util.TreeMap;
+
+public class CircularTreeMap<K, V> extends TreeMap<K, V> {
+
+	private static final long serialVersionUID = 1L;
+
+	private final int limit;
+
+	public CircularTreeMap(int limit) {
+		this.limit = limit;
+	}
+
+	@Override
+	public V put(K key, V value) {
+		V result = super.put(key, value);
+		if (super.size() > this.limit) {
+			this.removeEldest();
+		}
+		return result;
+	}
+
+	private void removeEldest() {
```

Comment:
TreeMap sortiert die Elemente, bei LocalDateTime sollte es problemlos funktionieren, wenn man sie mit anderen Keys benutzt,  würde z.B folgender Test fehlschlagen:

	@Test
	public void testAnotherOrder() {
		CircularTreeMap<String, String> m = new CircularTreeMap<>(3);
		m.put("4", "four");
		m.put("3", "three");
		m.put("2", "two");
		m.put("1", "one");
		
		Set<String> ks = m.keySet();
		Iterator<String> i = ks.iterator();
		assertEquals("3", i.next());
		assertEquals("2", i.next());
		assertEquals("1", i.next());
	}

Man könnte die Keys festlegen, also so in der Art: public class CircularTreeMap<V> extends TreeMap<LocalDateTime, V>